docs/source/contributing/roadmap.md

```diff
@@ -0,0 +1,46 @@
+# The repo2docker roadmap
+
+This roadmap collects "next steps" for the project. Each step should fit into
+one of the following three categories:
+
+* **now**, very concrete/actionable step that is ready for someone to pick
+it up and work on
+* **soon**, less concrete/actionable step that is going to happen soon,
+discussions around the topic are coming close to an end at which point it can
+move into the "now" category
+* **later**, abstract ideas or tasks, need a lot of discussion or
+experimentation to shape the idea so that it can be discussed. Should also
+contain concrete/actionable steps that have been postponed on purpose
+(these are steps that could be in "now" but the decision was taken to work on
+them later)
+
+The roadmap will get updated as time passes (next review by 1st December).
+This means this list should not be exhaustive, it should only represent
+the "top of the stack" of ideas. It should
+not function as a wish list or collection of feature requests. For those please
+create a [new issue](https://github.com/jupyter/repo2docker/issues/new).
+
+The roadmap should give the reader an idea of what is happening next, what needs
+input and discussion before it can happen and what has been postponed.
+
+
+> **Guiding thought for the next months**
+> Repo2docker is a dependable tool used by humans that reduces the complexity of creating the environment in which a piece of software can be executed.
+
+## Now
```

Comment:
What if we didn't list anything under "now", and instead we provide a link to the issue list for the next tagged release. With language like "repo2docker collects actionable and specific items for its next release at this link: XXX"

Reply:
or we can list things that are only too large to be fit under a single issue

Reply:
(e.g., under "now" we can say something like:

"repo2docker keeps track of actionable items in GitHub issues. To see a list of items that are assigned to upcoming releases of repo2docker, see [the repo2docker milestones page](https://github.com/jupyter/repo2docker/milestones)"

Reply:
Not sure. It feels like we'd be sending the user on a trip from GitHub to docs to GitHub :-/

If we use tags/boards then why not use them for all stages and get rid of the file beyond having a short sentence explaining which tags/board to look at?

A downside of having tags would be that we will use them liberally and then have to spend time shuffling them around when revisiting the roadmap (a lot more clicking then moving sentences in a text file). I'd also avoid combining releases with the roadmap stages (now==next release). I think it is fine to have something in Now that won't be in the next release. I'd also not want to try and predict which release something will be in and instead prefer "in some future release".

Reply:
that's a fair point - in my mind, I imagine that the roadmap is for more "abstract goals" of the project, and then anything that becomes concrete enough to have a specific issue is then turned into an issue and added to a milestone. I'm also happy with keeping a list in the "now" section though just worried that since this will be the most dynamic section, it'll be out-of-date most quickly

Reply:
I think we should keep the roadmap sentences somewhat abstract which means they aren't a milestone-able thing ("increase test coverage") and hopefully it won't be totally out of date all the time because it is somewhat abstract :-/

I am conflicted because up-to-dateness is important but the tools we use to reduce the burden of keeping things up-to-date are built into GitHub (auto close, tags, PRs, milestones) and we set out to make a roadmap for people who don't spend their life inside GH but we also don't want to go back to a world where we spend a lot of effort on keeping our paperwork up to date :-/

Reply:
I'm all for that. If we add language to the roadmap like "these are intentionally abstract, and if you'd like to see more specific and actionable topics, check out the issues/milestones" then I'm +1 with the structure

Reply:
What do you think of the new L6++ ?

Reply:
I think it's a good place to start, and we can see how it goes in the coming weeks. WDYT?